docs/csharp/reference-semantics-with-value-types.md

```diff
@@ -89,16 +89,34 @@ There are several ways in which the compiler ensures that the read-only
 nature of an `in` argument is enforced.  First of all, the called method
 can't directly assign to an `in` parameter. It can't directly assign
 to any field of an `in` parameter. In addition, you cannot pass
-an `in` parameter to any method demanding the `ref` or `out` modifier.
-The compiler enforces that the `in` argument is a readonly variable. You
-can call any instance method that uses pass-by-value semantics. In
+an `in` parameter to any method using the `ref` or `out` modifier.
+The compiler enforces that the `in` argument is a readonly variable.
```

Comment:
This restriction also applies to fields if parameter is a struct. Basically if a struct variable is readonly its fields are readonly, recursively. Not sure how to say it better.
Basically `inParam.field1.field2.field3` cannot be assigned to, passed by `ref`, etc... , just the same as `inParam` itself if `inParsm`, `field1`, `field2` are all structs and `inParam` is an `in` parameter.